features/distributed/coordination/api/src/main/java/org/opennms/features/distributed/coordination/api/RoleChangeHandler.java

```diff
@@ -0,0 +1,52 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2018 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2018 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.features.distributed.coordination.api;
+
+/**
+ * Defines the callbacks invoked by a {@link DomainManager} when a registrant becomes ACTIVE or STANDBY.
+ */
+public interface RoleChangeHandler {
```

Comment:
Just a thought: Is it possible to change this to a `@FunctionalInterface` by having something like `roleChanged(Role newRole, String domain)`?

Reply:
Yes, good suggestion. This also helps in case we were to add additional roles since we will have already forced the existing role handling logic to be in one place for every client.